dashboard/test/factories/factories.rb

```diff
@@ -72,7 +72,7 @@
 
   factory :user do
     birthday Time.zone.today - 21.years
-    email {("#{user_type}_#{(User.maximum(:id) || 0) + 1}@code.org")}
+    email {("#{user_type}_#{SecureRandom.uuid}@code.org")}
```

Comment:
My first attempt at this PR [used a `sequence` here](https://github.com/code-dot-org/code-dot-org/pull/36555/files#diff-286a7ecb55a740b99e7f8b0fb50a8b8dL75) to attempt to reduce the chance of collision. I suspect - but am so far unable to prove - that this is what caused the `Validation failed: Email has already been taken` error we saw.

Because I am so far unable to repro this locally, I can't be certain that using a UUID rather than a sequential ID here will solve the problem, but I am nonetheless pretty confident.

Reply:
it might help to know what email address was already taken when these tests failed.

Reply:
`teacher_5@code.org`, at least when I run just the relevant subset of tests via `bundle exec runner.rb -l -f features/plc/pd/* --eyes --html`

Reply:
whoa, any idea how a UI test is depending on a factory method? I thought we only used them for unit tests.

Reply:
Looks like we use FactoryGirl pretty extensively in the PD UI tests:

https://github.com/code-dot-org/code-dot-org/blob/51706f35e5ec7d60bcfbe6c85d9dc38c7c385e5e/dashboard/test/ui/features/step_definitions/pd.rb#L36
https://github.com/code-dot-org/code-dot-org/blob/51706f35e5ec7d60bcfbe6c85d9dc38c7c385e5e/dashboard/test/ui/features/step_definitions/pd.rb#L48-L50
https://github.com/code-dot-org/code-dot-org/blob/51706f35e5ec7d60bcfbe6c85d9dc38c7c385e5e/dashboard/test/ui/features/step_definitions/pd.rb#L104-L109
(and lots more)

Reply:
mmmm interesting! the first 2 calls look ok because they are using their own mechanism to generate a unique email, so they wouldn't touch the uuid logic you added. However, this one call does not specify an email and would therefore hit your uuid logic: https://github.com/code-dot-org/code-dot-org/blob/5756d75db4d9e85d6ffb7cb02f595f360ea70339/dashboard/test/ui/features/step_definitions/pd.rb#L421

Can you double-check my reasoning here, and say whether you think it might be worth changing the user factory email back to using `sequence` and making the line linked above specify an `email`? I would be willing to open the PR for this.

Reply:
I'm not seeing the value in that solution; in particular, it seems to me that you shouldn't _need_ to specify an email in order for the factory to work.

Reply:
Good point. This solution sounds good, now that we know why `sequence` wasn't working, because it lets UI tests rely on factories for generating unique emails. Thanks for your willingness to investigate!

Reply:
Thank you for your thoroughness!